app/src/main/java/com/example/Egida/presentation/viewModel/MainViewModel.kt

```diff
@@ -0,0 +1,67 @@
+package com.example.Egida.presentation.viewModel
+
+import android.text.TextUtils
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.Egida.Dependencies
+import com.example.Egida.domain.entity.User
+import com.example.Egida.domain.useCase.UserUseCase
+import kotlinx.coroutines.launch
+
+
```

Comment:
You are reusing MainViewModel in all fragments. It's not so bad, but you definitely could split it into separate ViewModels according to logic usage.